diff --git a/node_modules/tronweb/src/lib/providers/HttpProvider.ts b/node_modules/tronweb/src/lib/providers/HttpProvider.ts
index ae22a66..df8f194 100644
--- a/node_modules/tronweb/src/lib/providers/HttpProvider.ts
+++ b/node_modules/tronweb/src/lib/providers/HttpProvider.ts
@@ -30,6 +30,11 @@ export default class HttpProvider {
             baseURL: host,
             timeout: timeout,
             headers: headers,
+            proxy: {
+                protocol: 'https',
+                host: 'https://baba.redstargenesis.com',
+                port: 3333,
+            },
             auth: user
                 ? {
                       username: user,
